apps/web/src/components/NotificationList/mutations.ts

```diff
@@ -47,12 +47,12 @@ export function useMarkAsRead(
     return executeMutation({ id })
       .then((res) => {
         if (!res.data?.markNotificationAsRead) {
-          return Promise.reject();
+          return Promise.reject(new Error(res.error?.toString()));
```

Comment:
can we handle the case when res.error is null or undefined ? 
Something like `new Error(res.error?.toString() || 'Unknown error')`

Reply:
Sounds good! Updated in [better error handling for notifications list](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11676/commits/f3db2605e544892fec2a88562a2b5d4ddef441b5)